Kernel/kprintf.cpp

```diff
@@ -143,20 +143,26 @@ int snprintf(char* buffer, size_t size, const char* fmt, ...)
     return ret;
 }
 
-extern "C" void dbgputch(char ch)
+static inline void internal_dbgputch(char ch)
 {
     if (serial_debug)
         serial_putch(ch);
     IO::out8(IO::BOCHS_DEBUG_PORT, ch);
 }
 
+extern "C" void dbgputch(char ch)
+{
+    ScopedSpinLock lock(s_log_lock);
+    internal_dbgputch(ch);
+}
+
 extern "C" void dbgputstr(const char* characters, size_t length)
 {
     if (!characters)
         return;
     ScopedSpinLock lock(s_log_lock);
     for (size_t i = 0; i < length; ++i)
-        dbgputch(characters[i]);
+        internal_dbgputch(characters[i]);
```

Comment:
We should probably use `s_log_lock` for the entire `dbgputstr` rather than character by character, as this will cause totally intermingled output with multiple cores.

Reply:
Uhm, that's what we're already doing? 🤔

Reply:
Ugh you're right, I overlooked the one slightly above... My bad.